### DISCUSSION TITLE
Is there any windows version available for mint-lang

### DISCUSSION THREAD

**Discussion by Muhammad-Sarfaraz** [mint-lang/mint#400] [Answered] [answered]
Hi there,
Was looking for a windows version of mint-lang,but couldn't found it.Is there any windows version available.
Or how can I run this on windows platform.

**Comment by nailuj29** [comment, marked as answer]
Crystal (the programming language Mint is written in) does not yet support windows. You can use docker or WSL until it does support windows.

**Reply by frnco** [reply]
Worth additions:
Windows is considered "Tier 3" on Crystal's Platform Support, which means some stuff works while some stuff doesn't work:
https://crystal-lang.org/reference/platform_support.html

I've been using Crystal for a while, long before I've even heard of Mint, and Windows support has been in the roadmap for quite a while, but it only gained some traction recently.

As per Crystal's roadmap (https://github.com/crystal-lang/crystal/wiki/Roadmap), windows support is in the works and the roadmap lists it's completion as a target, and it's being tracked at this issue: https://github.com/crystal-lang/crystal/issues/5430 .

It's hard to put a date on this kind of thing, but to anyone using Windows that is interested in Mint (And/or Crystal) I highly recommend investing a little time in getting Windows Subsystem for Linux (WSL) up and running. I realize dual booting can be a pain and very few people are willing to spend the necessary time to get used to something like Arch Linux (Love of my life BTW), but WSL is ridiculously easy and quick to set up, and gets you the most newcomer-friendly distro I know of, Ubuntu (Though they may have added more options but Debian-based stuff is just easy to get used to so go for it, if you learn to love then you can consider moving to Arch, Gentoo or whatever).

In effect, Crystal (And Mint by extension) DOES run on Windows, through WSL. Sure, you can't use the Windows APIs, but for quite a number of developers such as web developers that's irrelevant. Even when Crystal fully supports windows it will still be wiser for web developers to use WSL as the servers that will ultimately run the code will most likely be Linux machines, so when considering we're talking about Mint, to me this feels more like a matter of documentation and tooling than a matter of actual platform support. Anyone using Mint is coding for the web and, as such, should work from either WSL or some Unix-based OS (MacOS, Linux, BSD, whatever, anything that is POSIX-compatible).

BTW, small rant about my experience with WSL, last time I checked was a few years ago but even then the only thing that bothered me was local files in the Ubuntu "Machine" (WSL) had to be inside a ridiculously deep tree of folders and I couldn't figure out a way to map some reasonable folder to use in WSL, but as I mentioned that was years ago, and even then WSL did pretty much everything I threw at it, there were very few limitations to the system's functionality. I think this is pretty big, as for the first time in forever Windows users have a reasonable way to get some linux awesomeness in their lives, and I seriously think it's a waste that so many people ignore it, especially considering how hard it was for most of us to switch to using Linux as our main system.

**Comment by serge-hulne** [comment]
In theory, one could develop an app under Linux on Windows (using an emulator) and then copy the resulting /dist directory to windows (in theory the set of HTML, CSS, JS files in /dist is platform independent (since it is interpreted by a browser)).
